utils/update_checkout/tests/scheme_mock.py

```diff
@@ -118,6 +118,10 @@ def setup_mock_remote(base_dir, base_config):
         call_quietly(['git', 'symbolic-ref', 'HEAD', 'refs/heads/main'],
                      cwd=remote_repo_path)
         call_quietly(['git', 'clone', '-l', remote_repo_path, local_repo_path])
+        call_quietly(['git', 'config', 'user.name', 'apple_test'], 
+                     cwd=local_repo_path)
+        call_quietly(['git', 'config', 'user.email', 'apple_test@apple.com'], 
+                     cwd=local_repo_path)
```

Comment:
Will this overwrite local config for the user?

Reply:
Also, prefer not using `apple`. Maybe we can use `swift_test` and `no-reply@swift.org`

Reply:
> Will this overwrite local config for the user?

Yes

Reply:
>Will this overwrite local config for the user?

This should only update the configuration for the git repository which is created as part of this test unless I'm misunderstanding. From the [git config documentation](https://git-scm.com/docs/git-config)
>When writing, the new value is written to the repository local configuration file by default, and options --system, --global, --worktree, --file <filename> can be used to tell the command to write to that location (you can say --local but that is the default).

Also in the output of the error message:
`
Omit --global to set the identity only in this repository.
`

As for using the environment variables vs calling git config itself, i have no preference

Reply:
If those git config commands fail, they'd be ignored (because we are running them through `call_quietly`), and we'd still run into the original issue?

Reply:
@azharudd The commands will raise an exception
```python3
def call_quietly(*args, **kwargs):
    kwargs['stderr'] = subprocess.STDOUT
    try:
        subprocess.check_output(*args, **kwargs)
    except subprocess.CalledProcessError as e:
        raise CallQuietlyException(command=e.cmd, returncode=e.returncode,
                                   output=e.stdout) from e
```
`call_quietly` is intended to suppress stdout from what I'm gathering, not suppress exceptions (a bit confusing)